photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/RandomSearch.scala

```diff
@@ -68,65 +64,45 @@ class RandomSearch[T](
 
     require(n > 0, "The number of results must be greater than zero.")
 
-    // Load the initial observations
-    observations.init.foreach { case (candidate, value) =>
-      onObservation(candidate, value)
-    }
-
-    // Load the prior data observations
-    priorObservations.foreach { case (candidate, value) =>
-      onPriorObservation(candidate, value)
-    }
-
-    val (results, _) = (0 until n).foldLeft((List.empty[T], observations.last)) {
-      case ((models, (lastCandidate, lastObservation)), _) =>
+    if (observations.isEmpty) {
```

Comment:
Add a comment above saying that this is the first iteration.

Reply:
Strictly speaking, this is the first iteration when `observations` is empty. But in our code, `observations` is initialized explicitly, given by the initial model, the last argument in `runHyperparameterTuning`. So this if section is never executed. It is reserved only for the `find` function for API completeness.